dipy/sims/voxel.py

```diff
@@ -288,7 +299,265 @@ def multi_tensor(gtab, mevals, S0=100, angles=[(0, 0), (90, 0)],
+    codified respectively by 0, 1 and 2. Simulations are based on 
+    multicompartment models. Implementation of this function is based on 
+    equation 8 of:
+    
+    [1] R. Neto Henriques et al., "Exploring the 3D geometriy of the diffusion 
+        kurtosis tensor - Impact on the development of robust tractography 
+        procedures and novel biomarkers", NeuroImage (2015) 111, 85-99.
+    
+    Parameters
+    -----------
+    Dc : (6,) ndarray
+        Elements of the diffusion tensor for each individual compartment on the
+        multicompartmental model. Assumes the order Dxx, Dyy, Dzz, Dxy, Dxz, 
+        Dyz.
+    frac : float
+        Percentage of the contribution of each tensor. The sum of fractions
+        should be equal to 100%.
+    i : int
+        first element index (0 for x, 1 for y, 2 for z)
+    j : int
+        Second element index (0 for x, 1 for y, 2 for z)
+    k : int
+        Third element index (0 for x, 1 for y, 2 for z)
+    el: int
+        Fourth elements index (0 for x, 1 for y, 2 for z)
+    DT : (6,) ndarray
+        Elements of the global diffusion tensor.
+
+    Returns
+    --------
+    wijkl : Kurtosis tensor element
+    
+    """
+    if DT is None:
+        DT=np.zeros((3,3))
+        for i in range(len(frac)):
+            DT=DT+frac[i]*Dc[i]
+
+    wijkl=0
+    MD=(DT[0][0]+DT[1][1]+DT[2][2])/3
+    for f in range(len(frac)):
+        wijkl=wijkl+frac[f]*(Dc[f][i][j]*Dc[f][k][el]+Dc[f][i][k]*Dc[f][j][el]+
+                             Dc[f][i][el]*Dc[f][j][k])
+    wijkl=(wijkl-
+           DT[i][j]*DT[k][el]-DT[i][k]*DT[j][el]-DT[i][el]*DT[j][k])/(MD**2)
+    
+    return wijkl
+    
+    
+def single_diffkurt_tensors(gtab, S0=150, dt=None, kt=None, snr=None):
+    """ Simulated signal based on the diffusion and diffusion kurtosis 
+    tensors. Simulations are preformed assuming the DKI model.
+
+    Parameters
+    -----------
+    gtab : GradientTable
+        Measurement directions.
+    S0 : double,
+        Strength of signal in the presence of no diffusion gradient (also
+        called the ``b=0`` value).
+    dt : (6,) ndarray
+        Elements of the diffusion tensor.
+    kt : (15, ) ndarray
+        Elements of the diffusion kurtosis tensor.  
+    snr : float
+        Signal to noise ratio, assuming Rician noise.  None implies no noise.
+
+    Returns
+    --------
+    S : (N,) ndarray
+        Simulated signal based on the DKI model S=S0*exp(-b*D+1/6*b**2*D**2*K)
+    """
+    
+    if dt is None:
+        dt = diffusion_tensor
+
+    if kt is None:
+        kt = kurtosis_tensor
+        
+    A=dki_design_matrix(gtab)
+    
+    # define vector of DKI parameters
+    MD=sum(dt[0:3])/3
+    X=np.concatenate((dt,kt*MD*MD,np.array([np.log(S0)])),axis=0)
+    
+    # Compute signals based on the DKI model
+    S=np.exp(dot(A,X))
+
+    S = add_noise(S, snr, S0)
+
+    return S
+    
+def dki_design_matrix(gtab):
```

Comment:
It's fine to put this here for now, but we'll eventually need to have just one implementation of this, preferably in the reconst/dki.py file (which doesn't exist yet on this branch...), so we need to remember to remove this from here, and import it from there when we start working on merging these things together.